action.yml

```diff
@@ -44,6 +44,15 @@ runs:
       shell: bash
       run: |
         export MAVEN_OPTS="${{ inputs.jvm-options }}"
+        if [[ ! "$MAVEN_OPTS" =~ .*"-Xmx".* ]]; then
+          MEM_TOTAL=$(cat /proc/meminfo | grep MemTotal | awk '$3=="kB"{$2=int($2/1024); print $2}')
+          MEM_SYSTEM=1536
+          MEM_AVAILABLE=$(($MEM_TOTAL - $MEM_SYSTEM))
+
+          echo "Memory settings: total=${MEM_TOTAL}MB, system=${MEM_SYSTEM}MB, available=${MEM_AVAILABLE}MB"
+          export MAVEN_OPTS="$MAVEN_OPTS -Xmx${MEM_AVAILABLE}m"
```

Comment:
Are you sure that this will be exported globally and not only in action step shell?

Reply:
@rbrisuda it will be exported only in action shep stell but that's OK enough, isn't it?

Reply:
i don't think we need put it to GITHUB_ENV. I think better is have it only in action shell

Reply:
I have doubts that it will be available in next step, but if you tested it then ok. :)

Reply:
@rbrisuda: i will test it after merge :)

Reply:
> I have doubts that it will be available in next step, but if you tested it then ok. :)

Now I have double checked and I have realized this is not a single setp, but a composite one with multiple bash steps.. 

**I think Rudo is right, this probably will not work.**

Please test it by placing `echo "Maven options: $MAVEN_OPTS"` just before command `mvn \          -Pgithub,${{ inputs.profiles }} \` if those options are available. 

If not, merge steps `Maven options` `Run maven command` to the one shell.

Reply:
@mmichna I think every step is own shell instance so you are exporting it just for that step, check setup-maven action, that is reason why there is not used export but it is stored in GITHUB_ENV file, which is exactly for this purpose. here it is: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable

Reply:
done